mteb/tasks/MAIR/eng/MAIR.py

```diff
@@ -0,0 +1,88 @@
+)
+
+_MAIR_CITATION = """@inproceedings{Sun2024MAIR,
+  title={MAIR: A Massive Benchmark for Evaluating Instructed Retrieval},
+  author={Weiwei Sun and Zhengliang Shi and Jiulong Wu and Lingyong Yan and Xinyu Ma and Yiding Liu and Min Cao and Dawei Yin and Zhaochun Ren},
+  booktitle={EMNLP},
+  year={2024},
+}"""
+
+
+def get_metadata(task_name):
+    return TaskMetadata(
+        name="MAIR-" + task_name,
+        description="Recent information retrieval (IR) models are pre-trained and instruction-tuned on massive datasets and tasks, enabling them to perform well on a wide range of tasks and potentially generalize to unseen tasks with instructions. However, existing IR benchmarks focus on a limited scope of tasks, making them insufficient for evaluating the latest IR models. In this paper, we propose MAIR (Massive Instructed Retrieval Benchmark), a heterogeneous IR benchmark that includes 126 distinct IR tasks across 6 domains, collected from existing datasets. We benchmark state-of-the-art instruction-tuned text embedding models and re-ranking models. Our experiments reveal that instruction-tuned models generally achieve superior performance compared to non-instruction-tuned models on MAIR. Additionally, our results suggest that current instruction-tuned text embedding models and re-ranking models still lack effectiveness in specific long-tail tasks.",
+        reference="https://github.com/sunnweiwei/MAIR",
+        dataset={
+            "path": "MAIR-Bench/MAIR-Queries",
+            "revision": "7d24eac886a6ae6653a6b67433e1c302cb0e9ac6",
+        },
+        type="Retrieval",
+        category="s2p",
+        modalities=["text"],
+        eval_splits=TASK2SPLIT.get(task_name, []),
+        eval_langs=["eng-Latn"],
+        main_score="ndcg_at_10",
+        date=("2023-07-10", "2024-11-10"),
+        domains=["Web"],
+        task_subtypes=["Question answering"],
+        license="mit",
+        annotations_creators="expert-annotated",
+        dialect=[],
+        sample_creation="found",  # queries LLM generated, corpus samples are found (extracted from S2ORC)
+        bibtex_citation=_MAIR_CITATION,
+    )
+
+
+def load_data(self, **kwargs):
+    if self.data_loaded:
+        return
+    self.corpus, self.queries, self.relevant_docs = {}, {}, {}
+    queries_path = self.metadata_dict["dataset"]["path"]
+    docs_path = self.metadata_dict["dataset"]["path"].replace("-Queries", "-Docs")
```

Comment:
Can you place queries and docs in same repo?

Reply:
Thanks for the feedback. To keep Q/D in one repo, I could create a separate repo for each task. 

But is that necessary? I think having two repo for Q and D would be easier to manage  than having over hundreds of repo for each task.

I can still proceed with the separate repo if required.

Reply:
You can create different splits for queries and documents in the same repo

Reply:
Thanks, I see.

One issue is that the data in MAIR has a two-level structure: task → subtasks, as some tasks contain multiple subtasks (e.g., IFEval, SWE-Bench). It’s tricky to maintain this structure without flattening it into a single repository.

And I think people may not need to download all the data if they’re only interested in evaluating a few specific tasks. So, if we still need to put Q and D in a single repo, the best way might be to generate 126 separate repo (for each task).

Or do you have any other suggestions?